### PR TITLE
Assignment 2 Submission

### PR DESCRIPTION
**Pull request by Nematodes**
This is the baseline calculator application.

It handles basic arithmetic operators, negative numbers, and most edge cases (all of the ones I could find with a moderate amount of effort.)

It also contains comprehensive arithmetic unit tests, as well as the string unit tests.

I may (unlikely) update the calculator's functionality in a later commit.

### CODE REVIEW

File: Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+                    if (!isErrorThrown)
+                    {
+                        Console.WriteLine($"Result: {equationResult}{Environment.NewLine}");
+                    }
+                }
+            }
+        }
+
+        static int getOperand(string equationString, int startIndex, out int operandEndStringPosition)
```
Comment (Keboo) [inline Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs]: Always specify protection modifiers for your methods. In this case `private`.

File: Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                return 0;
+            }
+
+            if (equationString[stringPosition] == '-')
+            {
+                isOperandNegative = true;
+                stringPosition++;
+
+                // Admittedly an inelegant solution (large "not equal to" chain), but it is fully functional
+                if (stringPosition >= equationString.Length || (equationString[stringPosition] != '0' && equationString[stringPosition] != '1' && equationString[stringPosition] != '2' && equationString[stringPosition] != '3' && equationString[stringPosition] != '4' && equationString[stringPosition] != '5' && equationString[stringPosition] != '6' && equationString[stringPosition] != '7' && equationString[stringPosition] != '8' && equationString[stringPosition] != '9'))
```
Comment (Keboo) [inline Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs]: I appreciate the comment here. There are a couple ways this could have been done. One is to leverage the `char.IsDigit` method.

File: Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            }
+            
+            while (equationString[stringPosition] != '+' && equationString[stringPosition] != '-' && equationString[stringPosition] != '*' && equationString[stringPosition] != '/' && equationString[stringPosition] != ' ')
+            {
+                checked
+                {
+                    try
+                    {
+                        tempOperand *= 10;
+                        tempOperand += Convert.ToInt32(equationString[stringPosition].ToString());
```
Comment (Keboo) [inline Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs]: It may have been simpler to build up a string containing all of the digits and then do the conversion at the end, but this works.

File: Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                    }
+                    catch (OverflowException)
+                    {
+                        // Minor quirk: ints are allowed to contain the value -2147483648 (int.MinValue), but doing arithmetic on that value triggers an overflow exception, even if it doesn't overflow (such subtracting 0 or adding 1)
```
Comment (Keboo) [inline Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.cs]: This comment has a couple issues. Only certain operations on `int.MinValue` will cause the overflow exception.  
`int.MinValue - 0` and `int.MinValue + 1` do not cause overflow exceptions.

File: Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.csproj [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{B0A80261-ECA6-4873-A948-5062A1748B4A}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <RootNamespace>ConsoleCalculator</RootNamespace>
+    <AssemblyName>ConsoleCalculator</AssemblyName>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Comment (Keboo) [inline Assignment2/ConsoleCalculator/ConsoleCalculator/ConsoleCalculator.csproj]: It looks like you may have created this console app in Visual Studio and picked the wrong framework. This is the old csproj format (you can compare it to what the .csproj for you unit tests looks like). When creating projects in Visual Studio make sure you select the console project from within the .NET Core section, not the Windows Desktop version.

**Comment by Keboo** [comment]
Nice work. Next time just make sure you are using .NET Core for all of your projects.
